### PR TITLE
ResourceQuotaTest added

### LINKED ISSUE #1381: Test coverage missing for multiple resources
The following resources don't have any test coverage

  - [ ] Endpoints
  - [ ] PersistentVolumeClaim
  - [ ] PersistentVolumes
  - [ ] SubjectAccessReview
  - [ ] PodPreset
  - [ ] HorizontalPodAutoscaler
  - [ ] Namespace
  - [ ] ResourceQuota

### PR DESCRIPTION
**Pull request by kunal-kushwaha**
Fixes #1381
Test coverage provided for ResourceQuota.

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceQuotaTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    assertNotNull(resourcequota);
+
+    deployment = client.apps().deployments().inNamespace("myspace").withName("deployment").get();
+    assertNotNull(deployment);
+  }
+
+  @Test
+  public void testResources() {
+    Config config = new ConfigBuilder().build();
+    try (final KubernetesClient client = new DefaultKubernetesClient(config)) {
```
Comment (rohanKanojia) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceQuotaTest.java]: Why are you not mocking server here? We don't have a cluster running during unit tests, however you can do in in integration tests(`kubernetes-itests/` module).

Comment (kunal-kushwaha) [reply kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceQuotaTest.java]: Moving to itests/

Comment (kunal-kushwaha) [reply kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceQuotaTest.java]: Hmm, moving to itests/

**Pull request merged**